utils/funcexp/functor_str.h

```diff
@@ -104,6 +105,18 @@ class Func_Str : public Func
 				return fp->data()->getStrVal(row, isNull);
 			break;
 		}
+        // Strip off the '+' that %g puts before the exponent symbol
+        dest = src = buf;
+        while(*src != '\0')
+        {
```

Comment:
does it make sense to scan the whole string - couldn't you just check the first character and then adjust the pointer if it's plus?